src/CommunityToolkit.Maui/Views/GravatarImageSource.shared.cs

```diff
@@ -0,0 +1,176 @@
+
+	/// <summary>The backing store for the <see cref="Image" /> bindable property.</summary>
+	public static readonly BindableProperty ImageProperty = BindableProperty.Create(nameof(Image), typeof(DefaultImage), typeof(GravatarImageSource), defaultValue: GravatarImageSourceDefaults.Defaultimage, propertyChanged: OnDefaultImagePropertyChanged);
+
+	int gravatarSize;
+
+	/// <summary>Initializes a new instance of the <see cref="GravatarImageSource"/> class.</summary>
+	public GravatarImageSource()
+	{
+		Stream = new Func<CancellationToken, Task<Stream>>(cancelationToken => DownloadStreamAsync(Uri, cancelationToken));
+		Uri = new Uri(defaultGravatarImageAddress);
+	}
+
+	/// <summary>On property changed.</summary>
+	/// <param name="propertyName">Property name.</param>
+	protected override void OnPropertyChanged(string propertyName)
+	{
+		if (propertyName is not null && (propertyName == CacheValidityProperty.PropertyName || propertyName == CachingEnabledProperty.PropertyName))
+		{
+			OnUriChanged();
+		}
+
+		base.OnPropertyChanged(propertyName);
+	}
+
+	/// <summary>Gets or sets a <see cref="TimeSpan"/> structure that indicates the length of time after which the image cache becomes invalid.</summary>
+	public TimeSpan CacheValidity
+	{
+		get => (TimeSpan)GetValue(CacheValidityProperty);
+		set => SetValue(CacheValidityProperty, value);
+	}
+
+	/// <summary>Gets or sets a Boolean value that indicates whether caching is enabled on this <see cref="GravatarImageSource"/> object.</summary>
+	public bool CachingEnabled
+	{
+		get => (bool)GetValue(CachingEnabledProperty);
+		set => SetValue(CachingEnabledProperty, value);
+	}
+
+	/// <summary>Gets or sets the email address.</summary>
+	public string? Email
+	{
+		get => (string)GetValue(EmailProperty);
+		set => SetValue(EmailProperty, value);
+	}
+
+	/// <summary>Gets or sets the default image.</summary>
+	public DefaultImage Image
+	{
+		get => (DefaultImage)GetValue(ImageProperty);
+		set => SetValue(ImageProperty, value);
+	}
+
+	/// <summary>Gets a value indicating whether the control URI is empty.</summary>
+	public override bool IsEmpty => Uri == null;
+
+	/// <summary>Gets or sets the URI for the image to get.</summary>
+	[System.ComponentModel.TypeConverter(typeof(UriTypeConverter))]
+	public Uri Uri { get; private set; }
+
+	/// <summary>Gets or sets the control size property.</summary>
+	int GravatarSize
+	{
+		get => gravatarSize;
+		set => gravatarSize = Math.Clamp(value, 1, 2048);
+	}
+
+	/// <summary>Returns the Uri as a string.</summary>
+	/// <returns>String of the URI.</returns>
+	public override string ToString()
+	{
+		return $"Uri: {Uri}, Email:{Email}, Size:{GravatarSize}, Image: {DefaultGravatarName(Image)}, CacheValidity:{CacheValidity}, CachingEnabled:{CachingEnabled}";
+	}
+
+	/// <summary>On parent set.</summary>
+	protected override void OnParentSet()
+	{
+		object element = Parent;
+		base.OnParentSet();
+		if (element is Microsoft.Maui.Controls.Image parentImage)
```

Comment:
Do we need to consider `Button` here? Or does that end up with a `Parent` of `Image` (I haven't checked yet myself)?

Reply:
@bijington I need to expand the unit tests to cover all controls that provide `ImageSource`, to ensure completness.

Reply:
That includes making sure that this check also operates as expected. (More investigation needed).

Reply:
I've changed the code to eliminate type specific checks.